### PR TITLE
[locator] add priority per group

### LINKED ISSUE #33096: Providing more info and/or better sorting of locator result groups
There is currently no details about result groups in locator results. It's a simple QString that is used as group text.

Giving more information on the group while creating the results doesn't look right (feeding many times the same info) nor it sounds possible to do it while keeping the API compatibility.

I would rather propose to provide more information in the locator filter itself. We could have a map 
`GroupId, ResultGroupInfo` holding all the groups details GroupId as a string, `ResultGroupInfo` being a struct probably containing display string, description (for second column in the results window) and order.

To preserve current functionality, if a filter doesn't implement the map of groups info, the string given 
 when creating the result would be used both as group id and as display string (description would be empty, order would be kept as appearing order).

### PR DESCRIPTION
**Pull request by 3nids**
This adds the possibility for a locator filter to define the ordering of groups within a filter.

fixes #33096

### CODE REVIEW

File: src/core/locator/qgslocatorfilter.h [hunk 1]
```diff
@@ -97,6 +97,14 @@ class CORE_EXPORT QgsLocatorResult
       */
     QString group;
 
+    /**
+     * Define the scrore of the group to allow filtering.
```
Comment (nyalldawson) [inline src/core/locator/qgslocatorfilter.h:101]:
```suggestion
     * Specifies the score of the group to allow filtering.
```

File: src/core/locator/qgslocatorfilter.h [hunk 2]
```diff
@@ -97,6 +97,14 @@ class CORE_EXPORT QgsLocatorResult
...
+     * Score must be positive, higher score are shown first.
```
Comment (nyalldawson) [inline src/core/locator/qgslocatorfilter.h:102]:
```suggestion
     * Score must be positive, higher scores are shown first.
```

File: src/core/locator/qgslocatorfilter.h [hunk 3]
```diff
@@ -97,6 +97,14 @@ class CORE_EXPORT QgsLocatorResult
...
+     * If not defined the sorting of groups is made alphabetically.
```
Comment (nyalldawson) [inline src/core/locator/qgslocatorfilter.h:103]:
```suggestion
     * If the score is 0 then the sorting of groups is made alphabetically.
```

File: src/core/locator/qgslocatormodel.h [hunk 4]
```diff
@@ -62,7 +63,9 @@ class CORE_EXPORT QgsLocatorModel : public QAbstractTableModel
       ResultFilterPriority SIP_MONKEYPATCH_COMPAT_NAME(ResultFilterPriorityRole), //!< Result priority, used by QgsLocatorProxyModel for sorting roles.
       ResultScore SIP_MONKEYPATCH_COMPAT_NAME(ResultScoreRole), //!< Result match score, used by QgsLocatorProxyModel for sorting roles.
       ResultFilterName SIP_MONKEYPATCH_COMPAT_NAME(ResultFilterNameRole), //!< Associated filter name which created the result
-      ResultFilterGroupSorting SIP_MONKEYPATCH_COMPAT_NAME(ResultFilterGroupSortingRole), //!< Group results within the same filter results
+      ResultFilterGroupSorting SIP_MONKEYPATCH_COMPAT_NAME(ResultFilterGroupSortingRole), //!< Deprecated since QGIS 3.40 no longer used
```
Comment (nyalldawson) [inline src/core/locator/qgslocatormodel.h:66]:
```suggestion
      ResultFilterGroupSorting SIP_MONKEYPATCH_COMPAT_NAME(ResultFilterGroupSortingRole), //!< \deprecated QGIS 3.40. No longer used.
```

See https://github.com/qgis/QGIS/pull/58552

**Comment by 3nids** [comment]
if no objection, I'll merge it today

**Pull request merged**